### PR TITLE
[Feature-1647][dlink-admin] Yarnsession support applicationId and resoucemanager config 

### LINKED ISSUE #1647: [Feature][dlink-admin]  Yarnsession support applicationId and resoucemanager config
### Search before asking

- [X] I had searched in the [issues](https://github.com/DataLinkDC/dinky/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Currently, the dlink session mode does not support the applicationId option, which is a bad thing, When the yarnsession fails and changes the address, the failure does not come out. So We support the yarnsession option

### Use case

support the yarnsession option

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by alexander1005**
## Purpose of the pull request
Yarnsession support applicationId and resoucemanager config 
## Brief change log
Yarnsession support applicationId and resoucemanager config 
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Comment by alexander1005** [comment]
https://github.com/DataLinkDC/dinky/issues/1647

File: dlink-admin/src/main/java/com/dlink/model/Cluster.java [hunk 1]
```diff
@@ -76,4 +76,21 @@ public static Cluster autoRegistersCluster(String hosts, String name, String ali
         cluster.setEnabled(true);
         return cluster;
     }
+    public String getJobManagerHost() {
+        if (type.equals("yarn-session")) {
+            if (StrUtil.isBlank(resourceManagerAddr) || StrUtil.isBlank(applicationId))
+                return jobManagerHost;
```
Comment (aiwenmo) [inline dlink-admin/src/main/java/com/dlink/model/Cluster.java]: Where is resourceManagerAddr defined?
Where is applicationId defined?

File: dlink-admin/src/main/java/com/dlink/model/Cluster.java [hunk 2]
```diff
@@ -76,4 +76,21 @@ public static Cluster autoRegistersCluster(String hosts, String name, String ali
...
+            ObjectNode applicationInstants = YarnUtils.
+                    getApplicationInstants(resourceManagerAddr, applicationId);
+            if (YarnUtils.getApplicationStatus(applicationInstants).equals("RUNNING")) {
+                this.status = 1;
```
Comment (aiwenmo) [inline dlink-admin/src/main/java/com/dlink/model/Cluster.java]: Replace `"RUNNING"` with `JobStatus.RUNNING.getValue()` .

File: dlink-admin/src/main/java/com/dlink/service/impl/ClusterServiceImpl.java [hunk 3]
```diff
@@ -123,10 +123,40 @@ public List<Cluster> listAutoEnable() {
 
     @Override
     public Cluster registersCluster(Cluster cluster) {
-        checkHealth(cluster);
+        if (cluster.getType().equals("yarn-session")){
+            if (StrUtil.isBlank(cluster.getResourceManagerAddr())
```
Comment (aiwenmo) [inline dlink-admin/src/main/java/com/dlink/service/impl/ClusterServiceImpl.java]: Replace `"yarn-session"` with `GatewayType.YARN_SESSION.getLongValue()` .

File: dlink-doc/sql/dinky.sql [hunk 4]
```diff
@@ -110,6 +110,8 @@ CREATE TABLE `dlink_cluster` (
                                  `cluster_configuration_id` int DEFAULT NULL COMMENT 'cluster configuration id',
                                  `task_id` int DEFAULT NULL COMMENT 'task ID',
                                  `enabled` tinyint(1) NOT NULL DEFAULT '1' COMMENT 'is enable',
+                                 `application_id` varchar(50) CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL COMMENT 'Job Manager Host',
+                                 `resource_manager_addr` varchar(100) CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL COMMENT 'Job Manager Host',
```
Comment (aiwenmo) [inline dlink-doc/sql/dinky.sql]: The comment of application_id is wrong.
The comment of resource_manager_addr is wrong.

File: dlink-web/src/locales/en-US/pages.ts [hunk 5]
```diff
@@ -494,6 +494,8 @@ export default {
   'pages.devops.lable.line.piontConfig.default': 'Default SavePoint',
   'pages.devops.lable.line.piontConfig.last': 'Latest SavePoint',
 
+  'pages.rc.cluster.resourceManagerAddr' : 'ResourceManger Addr',
+  'pages.rc.cluster.applicationId' : 'applicationId',
```
Comment (aiwenmo) [inline dlink-web/src/locales/en-US/pages.ts]: Replace `ResourceManger Addr` with `Resource Manger Address` .
Replace `applicationId` with `Application Id` .

File: dlink-web/src/pages/RegistrationCenter/ClusterManage/Cluster/components/ClusterForm.tsx [hunk 6]
```diff
@@ -45,6 +45,8 @@ const ClusterForm: React.FC<ClusterFormProps> = (props) => {
     id: props.values.id,
     name: props.values.name,
     alias: props.values.alias,
+    resourceManagerAddr: props.values.resourceManagerAddr,
+    applicationId: props.values.applicationId,
     type: props.values.type,
```
Comment (aiwenmo) [inline dlink-web/src/pages/RegistrationCenter/ClusterManage/Cluster/components/ClusterForm.tsx:50]: How to control the visibility of parameters?

**Review by aiwenmo — CHANGES_REQUESTED** [review]

File: dlink-admin/src/main/java/com/dlink/model/Cluster.java (hunk 1 above)
Comment (alexander1005) [reply dlink-admin/src/main/java/com/dlink/model/Cluster.java]: Yarn session's config StrUtil.isBlank(resourceManagerAddr) and  StrUtil.isBlank(applicationId) was satisfied  , will manage.
the old Yarn session  jobManagerHost config, dont manged ,that's Previous way.

resourceManagerAddr  applicationId in db columns

**Comment by alexander1005** [comment]
image
image

**Comment by Zzm0809** [comment]
actions is failed ， Please check you have to import related code

**Comment by Zzm0809** [comment]
**dlink-admin/src/main/java/com/dlink/model/Cluster.java** StrUtil is not imported

**dlink-admin/src/main/java/com/dlink/service/impl/ClusterServiceImpl.java** StrUtil is not imported

**dlink-admin/src/main/java/com/dlink/service/impl/ClusterServiceImpl.java** ObjectNode is not imported

**Review by aiwenmo — APPROVED** [review]
LGTM

**Comment by aiwenmo** [comment]
This is a good feature. Welcome to submit it to dev branch. thx.

**Pull request merged**